Gruntfile.js

```diff
@@ -357,14 +354,14 @@ module.exports = (grunt) => {
     const tempPackageJson = JSON.parse(JSON.stringify(packageJson));
     const keys = Object.keys(tempPackageJson.dependencies).concat(Object.keys(tempPackageJson.devDependencies));
 
-    const bumps = Bluebird.map(keys, (dep) => {
+    const bumps = keys.map((dep) => {
       return bumpDependency(tempPackageJson, dep);
     });
 
-    Bluebird.all(bumps).then(() => {
-      fs.writeFileSync('package.json', `${JSON.stringify(tempPackageJson, null, 2)}\n`);
-      grunt.log.writeln('Dependencies bumped, run npm install to install latest versions.');
-    }).then(() => { done(); }).catch(done);
+    Promise.all(bumps)
+      .then(() => fs.promises.writeFile('package.json', `${JSON.stringify(tempPackageJson, null, 2)}\n`))
+      .then(() => grunt.log.writeln('Dependencies bumped, run npm install to install latest versions.'))
+      .then(() => { done(); }).catch(done);
```

Comment:
If we don't care about the result, we should consider using `finally`.

```suggestion
      .finally(done);
```

Reply:
We don't care about the result but about the potential error in `.catch`.

Reply:
As far as I can see, the done function only accepts `false` to signal that the task has failed.

> Note that passing false to the done() function tells Grunt that the task has failed.

The following task for example succeeds because the done function is passed a truthy value:
```js
  grunt.registerTask('fail-async', '', function() {
    const done = this.async();
    Promise.reject("Failing from async")
      .then(() => { done(); })
      .catch(done);
  });
```
![image](https://user-images.githubusercontent.com/2564094/81312417-7ba36f00-903b-11ea-971e-1103f0b135a4.png)


To fail on promise reject, we should do something like this:
```js
  grunt.registerTask('fail-async', '', function() {
    const done = this.async();
    Promise.reject("Failing from async")
      .then(done)
      .catch((reason) => {
        grunt.log.error(reason);
        done(false);
      });
  });
```

![image](https://user-images.githubusercontent.com/2564094/81313007-2a47af80-903c-11ea-8094-32c5e288990f.png)

Reply:
Ah, yes of course grunt uses an other behaviour than node...
https://gruntjs.com/creating-tasks#why-doesn-t-my-asynchronous-task-complete
I will revisit all usages of `this.async()` then.

Reply:
Looks like it works as expected when using actual `Error` objects.
```js
  grunt.registerTask('fail-async', '', function() {
    const done = this.async();
    Promise.reject(new Error("Failing from async"))
      .then(() => { done(); })
      .catch(done);
  });
```

![image](https://user-images.githubusercontent.com/4009570/81316242-a4535500-908b-11ea-9895-151f7176c314.png)

https://gruntjs.com/api/inside-tasks#this.async